### PR TITLE
[Merged by Bors] - Feat(RingTheory/Ideal/Operations): Misc lemmas about smul actions on submodules

### PR DESCRIPTION
**Pull request by Shamrock-Frost**
Misc lemmas about smul actions on submodules.

---

The overall API of elements/sets/ideals acting on submodules kind of needs of an overhaul. There's a lot of stuff duplicated between these three actions, and it can be hard to find what you need because lemma names don't necessarily distinguish between the actions (eg `map_smul''` does not at all indicate it's about ideal-actions and not element-actions or set-actions). This is made worse by the fact that the actions of an ideal is defined in terms of map2, which only works over commutative rings.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Ideal/Operations.lean [hunk 1]
```diff
@@ -318,6 +318,11 @@ theorem map_smul'' (f : M →ₗ[R] M') : (I • N).map f = I • N.map f :=
       hfp ▸ f.map_smul r p ▸ mem_map_of_mem (smul_mem_smul hr hp)
 #align submodule.map_smul'' Submodule.map_smul''
 
+open Pointwise in
+theorem map_element_smul (r : R) (N : Submodule R M) (f : M →ₗ[R] M') :
```
Comment (jcommelin) [inline Mathlib/RingTheory/Ideal/Operations.lean:322]: I don't know if we have a naming convention for this, but I would expect
```suggestion
theorem map_pointwise_smul (r : R) (N : Submodule R M) (f : M →ₗ[R] M') :
```

**Comment by erdOne** [comment]
Maybe make them simp lemmas?

**Comment by Shamrock-Frost** [comment]
If we're marking `map_pointwise_smul` as `simp` then `map_smul''` should be marked that too. I'm not sure the other lemma should be `simp`, neither side seems "simpler"

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**